### ISSUE TITLE
Custom Trigger Boundary for popovers

### ISSUE THREAD

**Issue by ritz078** [adobe/react-spectrum#4599]
# 🙋 Feature Request

Right now, the popover position is calculated relative to the trigger element on basis of the `triggerRef` that we pass to the popover. There are situations where we want the boundary of the popover to be calculated relative to another element.

Taking the below example:

![image](https://github.com/adobe/react-spectrum/assets/5389035/eb5231e6-9a1a-44ca-bed5-1a8d69dca5d3)

In the above example, we want to open the popover to the left of the sidebar instead of just relative to the trigger element irrespective of where in the sidebar the element is. So I want to set the custom trigger boundary relative to the trigger parent which takes the full width. 

## 💁 Possible Solution

`usePopover` can accept a new property `positionAnchor` which can accept a HTML element. This element can be used to position the popover.

## 🔦 Context

I have many buttons inside the sidebar that act as trigger elements for menu, dropdown etc. We want to open the dropdown outside of the sidebar. 

We tried to set the `offset` by calculating the distance ourselves but we can't use popover's `placement` logic to place it correctly depending on the space available in other places.

## 💻 Examples

Figma uses a similar approach to open dropdowns in their sidebar

<img width="566" alt="Screenshot 2023-05-26 at 10 46 01 PM" src="https://github.com/adobe/react-spectrum/assets/5389035/17bfab0e-46e7-44b6-b6e2-31c96f9a4828">

**Comment by LFDanLu** [comment]
The team is looking to refactor much of the positioning code in which we'd see about including this option and more but in the meantime you could drop down to the lower level aria hooks like [useOverlayPosition](https://react-spectrum.adobe.com/react-aria/useOverlayPosition.html) instead of `usePopover` and leverage the `targetRef` prop it exposes.

**Comment by devongovett** [comment]
I don't think you should even need to drop down. `usePopover` passes the `triggerRef` through so it should work fine.

**Comment by ritz078** [comment]
I totally missed this. Sorry for that. `triggerRef` makes the whole element clickable which I do not want. I just want the position to be calculated wrt that element. Am I misunderstanding something?

**Comment by ritz078** [comment]
`usePopover` can accept an optional `targetRef` and fallback to `triggerRef` when not present. The code is here: https://github.com/adobe/react-spectrum/blob/d976513bfaa5bdc9cc2e5d66efd72470e0549146/packages/%40react-aria/overlays/src/usePopover.ts#L89

Will you be open for a PR ?

**Comment by devongovett** [comment]
No, `triggerRef` does not make the element clickable. It's only used for positioning.
